### PR TITLE
plugins: compile plugins for Windows games also on Linux, for Proton/Wine support

### PR DESCRIPTION
**Pull request by davidebeatrici**
Support for Wine was added in #2512.

### CODE REVIEW

File: plugins/mumble_plugin_utils.h [hunk 1]
```diff
@@ -49,4 +54,43 @@ static void escape(char *str, size_t size) {
...
+	// Append NUL character and return if there's space only for it
+	if (size == 1) {
+		dst[0] = '\0';
+		return 0;
+	}
+
+	// Get source string length and decrease it if necessary
+	size_t len = strnlen(src, size);
+	if (len >= size) {
+		len = size - 1;
```
Comment (Kissaki) [inline plugins/mumble_plugin_utils.h]: The method comment and maybe also this block comment should document that if size is smaller than src that only size - 1 characters will be copied.

Comment (davidebeatrici) [reply plugins/mumble_plugin_utils.h]:
```
// This function copies the number of chars specified.
// If "size" is smaller than "src", the number of chars to be copied corresponds to "size" - 1, as the last space available is required to NUL-terminate the string.
```

**Comment by davidebeatrici** [comment]
> What’s up with the include change?

The new header includes `mumble_plugin_win32.h` on Windows and `mumble_plugin_linux.h` on Linux, to eliminate the need of writing in every source file:
```c++
#ifdef WIN32
#include "../mumble_plugin_win32.h" // Include standard plugin header.
#else
#include "../mumble_plugin_linux.h" // Include standard plugin header.
#endif
```

> How is enabling this done? What changes?

This pull request changes `plugins.pri` so that all plugins are built on Linux and fixes compile errors caused by the use of non-standard functions.

File: plugins/mumble_plugin_utils.h (hunk 1 above)
Comment (rpmjava) [reply plugins/mumble_plugin_utils.h]: Hi, I don't know where can I make this question but I'll try here.
I'm new on Mumble but if I want to make changes in this service application is possible to make in an IDE running on Windows?
What IDE you're using to run mumble to debug?
I'm doing by this way, changin a file and then build the application after that, deploing in a Ubuntu server to test it. For me to do this by this way is very hard and I think there is an easy  way to do this. can you help me?

Comment (davidebeatrici) [reply plugins/mumble_plugin_utils.h]: Hi, I personally use Qt Creator on both Linux and Windows, because it's deeply integrated with Qt.

**Pull request merged**